lib/honeydew.ex

```diff
@@ -363,12 +363,15 @@ defmodule Honeydew do
     supervisor_opts =
       opts
       |> Keyword.get(:supervisor_opts, [])
-      |> Keyword.put_new(:id, {:queue, name})
+      |> Enum.into(%{})
+
+    spec = %{
+      id: {:queue, name},
+      start: {Honeydew.QueueSupervisor, :start_link,
+              [name, module, args, num, dispatcher, failure_mode, success_mode, suspended]},
```

Comment:
since `Honeydew.QueueSupervisor` is a supervisor, i believe you're going to need to add `type: :supervisor` to the spec.

Reply:
(also in various other places in the PR)